util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,24 @@
+package com.twitter.util
+
+/** Provides a clean, lightweight interface for working with a batched [[com.twitter.util.Future]] */
+class Batcher[In, Out] private[util](
+  executor: BatchExecutor[In, Out]
+)(
+  implicit timer: Timer
+) extends Function1[In, Future[Out]] { batcher =>
+  /**
+   * Enqueues requests for a batched [[com.twitter.util.Future]]
+   *
+   * Processes all enqueued requests after the batch size is reached
+   */
+  def apply(t: In): Future[Out] = executor.enqueue(t)
+
+  /** Immediately processes all unprocessed requests */
+  def flushBatch(): Unit = {
```

Comment:
Here's how I decided to address this:

I moved the synchronization inside a new method in the BatchExecutor called .flushNow().  So, Batcher.flushBatch() just calls BatchExecutor.flushNow()

This is good IMO because:
* A person new to the code will find all synchronization logic inside of one file, BatchExecutor.scala, and they will easily be able to see that all calls to BatchExecutor.flushBatch are wrapped in synchronize blocks.
* If that person still doesn't understand what's going on, the comments explaining that synchronization is required are in the same file, not a a different file.
* Batcher.scala stays simple

Does this address your concerns?